### PR TITLE
chore(deps): terraform-provider-grafana from v3.13.1 to v3.15.0

### PR DESCRIPTION
**Pull request by Duologic**
Update along with some housekeeping

- **chore(Makefile): grab provider version from go.mod**
- **chore(build): update build scripts to fix CI**
- **chore(build): remove pinned UPTEST_VERSION**
- **chore(deps): update terraform-provider-grafana from 3.13.1 to 3.14.0**
- **chore: make clean generate**
- **chore: tell dependabot to use conventional commits**
- **chore: push to 3.15.0**

See https://github.com/grafana/terraform-provider-grafana/releases for upstream changes.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ updates:
     directory: "/"
     schedule:
       interval: "weekly"
+    commit-message:
+      prefix: ci
```
Comment (Pokom) [inline .github/dependabot.yml:8]: Nice, I didn't know you could influence the commit message of dependabot updates.

File: Makefile [hunk 2]
```diff
@@ -8,8 +8,7 @@ export TERRAFORM_VERSION := 1.7.5
 
 export TERRAFORM_PROVIDER_SOURCE := grafana/grafana
 export TERRAFORM_PROVIDER_REPO := https://github.com/grafana/terraform-provider-grafana
-# UPGRADE THE go.mod also!
-export TERRAFORM_PROVIDER_VERSION := 3.13.1
+export TERRAFORM_PROVIDER_VERSION := $(shell grep terraform-provider-grafana go.mod | awk '{print $$2}' | sed 's/v//')
```
Comment (Pokom) [inline Makefile:11]: Validated the output as:

```
 grep terraform-provider-grafana go.mod | awk '{print $$2}' | sed 's/v//'
        github.com/grafana/terraform-proider-grafana/v3 v3.15.0
```

**Review by Pokom — APPROVED** [review]
👍🏻

File: Makefile (hunk 2 above)
Comment (Duologic) [reply Makefile:11]: In the local shell you have to remove the additional `$` in the awk statement, the double `$$` is an escape mechanism for Makefile.

**Pull request merged**